improver/precipitation_type/shower_condition_probability.py

```diff
@@ -0,0 +1,182 @@
+"""Plugin to construct a shower conditions probability"""
+
+from typing import Dict, Optional, Tuple
+
+import iris
+import numpy as np
+from iris.cube import Cube, CubeList
+from iris.exceptions import CoordinateNotFoundError
+
+from improver import PostProcessingPlugin
+from improver.metadata.constants import FLOAT_DTYPE
+from improver.metadata.utilities import (
+    create_new_diagnostic_cube,
+    generate_mandatory_attributes,
+)
+from improver.threshold import BasicThreshold
+from improver.utilities.cube_manipulation import collapse_realizations
+
+from .utilities import make_shower_condition_cube
+
+
+class ShowerConditionProbability(PostProcessingPlugin):
+    """Plugin to calculate the probability that conditions are such that
+    precipitation will be showery, based on input cloud amounts and the
+    convective ratio."""
+
+    def __init__(
+        self,
+        cloud_threshold: float = 0.5,
+        convection_threshold: float = 0.5,
+        model_id_attr: Optional[str] = None,
+    ) -> None:
+        """
+        Args:
+            cloud_threshold:
+                The fractional cloud coverage value at which to threshold the
+                cloud data; default 0.5.
+            convection_threshold:
+                The convective ratio value at which to threshold the convective
+                ratio data; default 0.5.
+            model_id_attr:
+                Name of the attribute used to identify the source model for
+                blending.
+        """
+        self.cloud_threshold = cloud_threshold
+        self.convection_threshold = convection_threshold
+        self.model_id_attr = model_id_attr
+        self.cloud_constraint = iris.Constraint(
+            cube_func=lambda cube: "cloud_area_fraction" in cube.name()
+        )
+        self.convection_constraint = iris.Constraint(
+            cube_func=lambda cube: "convective_ratio" in cube.name()
+        )
+
+    def _output_metadata(self, cube: Cube) -> Tuple[Cube, Dict]:
+        """
+        Returns template cube and mandatory attributes for result. The template
+        cube is modified to introduce an implied shower conditions threshold.
+
+        Args:
+            cube:
+                The cube to use as a template, and from which to extract
+                attributes for use in the new diagnostic cube.
+        Returns:
+            A tuple containing the template cube and attributes.
+        """
+        template = make_shower_condition_cube(cube)
+
+        attributes = generate_mandatory_attributes(
+            [cube], model_id_attr=self.model_id_attr
+        )
+        return template, attributes
+
+    def _extract_inputs(self, cubes):
+        """Extract the required input cubes from the input cubelist and check
+        they are as required."""
+        try:
+            (cloud,) = cubes.extract(self.cloud_constraint)
+            (convection,) = cubes.extract(self.convection_constraint)
+        except ValueError:
```

Comment:
While I like `try...except` clauses, I don't think they should be used if it is possible to detect the specific condition you want.
```suggestion
        cloud = cubes.extract(self.cloud_constraint)
        convection = cubes.extract(self.convection_constraint)
        if len(cloud) == 1 and len(convection) == 1:
            (cloud,) = cloud
            (convection,) = convection
        else:
```

Reply:
But python.... https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/

Reply:
Your link only says that you can, not that you should.
I think the point of exceptions is to highlight where and why code failed, which the code as it stands does. I guess it can stay as it is.